src/roll.c

```diff
@@ -0,0 +1,143 @@
+#include "data.table.h"
+#include <Rdefines.h>
+
+static SEXP rollmeanVectorRaw(SEXP tmp, SEXP this, R_len_t xrows, R_len_t thisk, SEXP thisfill, SEXP narm) {
```

Comment:
This function is wrapped into openmp from `rollmean` function. Is it fine to pass arguments as is or should be passed as pointers?

Reply:
You're doing well!  Good question.  There has to be an entry point taking SEXP if the function is callable from R; that's clear.  But then when we want to call that from other C code, it is inconvenience to have to construct the SEXP objects just to call that function.   So we have sometimes turned the entry point into a thin wrapper which then passes the regular C variables on to the internal C function.   Then again,  if the function is highly iterated, it may be either inline function or the algo changed to not use a function call at all.   So the answer is you can pass arguments, or C pointers or SEXP pointers,  each situation is different.

Reply:
I feel having an R function called from multi-threaded code is too dangerous. Constructing/destroying SEXPs will probably lead to memory corruption; but even simple access to SEXP api may have unpredictable consequences if R does something non-trivial under the hood...

Reply:
@st-pasha This comment feels a little out of place. I don't know why you're writing this here. I've written quite a bit on multi-threaded issues elsewhere in comments in this PR. So I'm not sure your point exactly here.

Reply:
Solved by using `void` instead of `SEXP`. Function saves results into `tmp` so don't have to return anything.